hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogSyncClient.java

```diff
@@ -178,6 +179,16 @@ private static GlueAsyncClient buildAsyncClient(HiveSyncConfig config) {
     }
   }
 
+  @Override
+  public String getTableName() {
+    return config.getStringOrDefault(GLUE_SYNC_TABLE_NAME, GLUE_SYNC_TABLE_NAME.getInferFunction().get().apply(config).get());
```

Comment:
Note to Reviewer: 
We do not instantiate or use the GlueCatalogSyncClientConfig class here and its an instance of HivesyncConfig. So, the infer func automatically does not kick in. so, I had to take this route. 

Same applies for DatahubSyncConfig as well.

Reply:
Looks like we always fetch the db and table name from the sync client from the sync tool, so I guess there is no need to infer the names again in these getters? Just initializes them well in the sync client constructor and return it here?

Reply:
Can we move the instantiation of the table name to the constructor, just like the databaseName?